packages/react-i18n/src/babel-plugin/babel-templates.ts

```diff
@@ -0,0 +1,71 @@
+import {TemplateBuilder} from '@babel/template';
+import Types from '@babel/types';
+
+import {
+  TRANSLATION_DIRECTORY_NAME,
+  getLocaleIds,
+  toArrayString,
+} from './shared';
+
+export function fallbackTranslationsImport(
+  template: TemplateBuilder<Types.ImportDeclaration | Types.ObjectExpression>,
+  {id, fallbackLocale},
+) {
+  return template(
+    `import ${id} from './${TRANSLATION_DIRECTORY_NAME}/${fallbackLocale}.json';`,
+    {
+      sourceType: 'module',
+    },
+  )() as Types.ImportDeclaration;
+}
+
+export function translationsImport(
+  template: TemplateBuilder<Types.ImportDeclaration | Types.ObjectExpression>,
+  {id},
+) {
+  return template(`import ${id} from './${TRANSLATION_DIRECTORY_NAME}';`, {
+    sourceType: 'module',
+  })() as Types.ImportDeclaration;
+}
+
+export function i18nCallExpression(
+  template: TemplateBuilder<Types.ImportDeclaration | Types.ObjectExpression>,
+  {
+    id,
+    fallbackID,
+    translationArrayID,
+    bindingName,
+    translationFilePaths,
+    fallbackLocale,
+  },
+) {
+  const translationArrayString = translationArrayID
+    ? translationArrayID
+    : toArrayString(
+        getLocaleIds({
+          translationFilePaths,
+          fallbackLocale,
+        }),
+      );
+
+  return template(
+    `${bindingName}({
+        id: '${id}',
+        fallback: ${fallbackID},
+        translations(locale) {
+          if (${translationArrayString}.indexOf(locale) < 0) {
```

Comment:
I think this would return false-positives for hyphenated locales.  e.g.:
* `hi` and `hi-IN` are used in our apps
* If we have only `translations/hi-IN.json`
* `"['hi-IN']".indexOf('hi')` would be a bug

Reply:
['hi-IN'].indexOf('hi') = -1
['hi-IN', 'hi'].indexOf('hi') = 1

I think that make sense??? huh??

Reply:
I can change it to use inclIded thou if that make more sense

Reply:
No, you're right.  I was forgetting that this is a template, and that `translationsArrayString` would be an actual array at the time of comparison.